lib/enmeshed/connector.rb

```diff
@@ -8,119 +8,131 @@ class Connector
+      def parse_result(response, klass)
+        json = JSON.parse(response.body).deep_symbolize_keys
 
-    # @return [klass, Array<klass>]
-    # @raise [ConnectorError] If the response contains an error or cannot be parsed.
-    def self.parse_result(response, klass)
-      json = JSON.parse(response.body).deep_symbolize_keys
+        if json.include?(:error)
+          raise ConnectorError.new(
+            "Enmeshed connector response contained error: #{json[:error][:message]}. " \
+            "Full response was: #{response.body}"
+          )
+        end
 
-      if json.include?(:error)
-        raise ConnectorError.new(
-          "Enmeshed connector response contained error: #{json[:error][:message]}. Full response was: #{response.body}"
-        )
+        parse_enmeshed_object(json[:result], klass)
+      rescue JSON::ParserError
+        raise ConnectorError.new("Enmeshed connector response could not be parsed. Received: #{response.body}")
       end
 
-      parse_enmeshed_object(json[:result], klass)
-    rescue JSON::ParserError
-      raise ConnectorError.new("Enmeshed connector response could not be parsed. Received: #{response.body}")
-    end
-    private_class_method :parse_result
-
-    # @return [klass, Array<klass>] The parsed object or array of objects.
-    def self.parse_enmeshed_object(content, klass)
-      if content.is_a?(Array)
-        content.map {|object| klass.parse(object) }
-      else
-        klass.parse(content)
+      # @return [klass, Array<klass>] The parsed object or array of objects.
+      def parse_enmeshed_object(content, klass)
+        if content.is_a?(Array)
+          content.map {|object| klass.parse(object) }
+        else
+          klass.parse(content)
+        end
       end
-    end
-    private_class_method :parse_enmeshed_object
 
-    # @return [Faraday::Connection] The connection to the enmeshed connector.
-    def self.conn
-      @conn ||= init_conn
-    end
-    private_class_method :conn
+      # @return [Faraday::Connection] The connection to the enmeshed connector.
+      # @raise [ConnectorError] If the connector is not configured as expected.
+      def connection
+        unless User.omniauth_providers.include?(:nbp) && CONNECTOR_URL.present? && API_KEY.present? \
+          && RelationshipTemplate::DISPLAY_NAME.present?
+          raise ConnectorError.new('NBP provider or enmeshed connector not configured as expected')
+        end
 
-    # @return [Faraday::Connection] A new connection to the enmeshed connector.
-    # @raise [ConnectorError] If the connector is not configured as expected.
-    def self.init_conn
-      if User.omniauth_providers.exclude?(:nbp) || CONNECTOR_URL.nil? || API_KEY.nil? || RelationshipTemplate::DISPLAY_NAME.nil?
-        raise ConnectorError.new('NBP provider or enmeshed connector not configured as expected')
+        Faraday.new(CONNECTOR_URL, headers:, request: {open_timeout: 1, read_timeout: 5, write_timeout: 5})
```

Comment:
We can use this block approach, too:

```suggestion
        Faraday.new(CONNECTOR_URL, headers:) do |faraday|
          faraday.options[:open_timeout] = 1
          faraday.options[:timeout] = 5
          
          faraday.adapter Faraday.default_adapter
        end
```

Reply:
Do you still prefer to move the definition of the request options into the block with the current version?

Reply:
I have a slight tendency towards that, since it would be in line with the other settings, e.g., as performed here:

https://github.com/openHPI/codeharbor/blob/aa987d69988165e55f916c7cb7e9e1025f2de9f5/app/services/task_service/check_external.rb#L43-L48